### ISSUE TITLE
custom_lint does not work with pub workspaces

### ISSUE THREAD

**Issue by Pante** [invertase/dart_custom_lint#298]
I have a [lint](https://github.com/forus-labs/forui/tree/experiment/lint/forui_internal_lints) that was working before migration to pub workspaces. After migrating to pub workspaces, the lints stopped working with the following log outputs:
* [forui's output](https://github.com/forus-labs/forui/blob/experiment/lint/forui/custom_lint.log)
* [forui_internal_lint/testbed's output](https://github.com/forus-labs/forui/blob/experiment/lint/forui_internal_lints/testbed/custom_lint.log)

It seems custom_lint is unable to find `package_config.json`s. Looking in the respective `.dart_tools` folders, there are `package_config_subset`s but no`package_config.json`s.

**Comment by Pante** [comment]
Any updates on this @rrousselGit? We think that custom_lint is a game-changer, and it's a shame that we can't use it with pub workspaces.

**Comment by rrousselGit** [comment]
No. I don't think I'll update Custom_lint for workspaces, as Google likely will release an official way to make plugins soon

**Comment by Pante** [comment]
I see, do you have any links to any design docs or other resources they have? Last I heard was them deprecating analyzer plugins altogether.

**Comment by lishaduck** [comment]
> I see, do you have any links to any design docs or other resources they have? Last I heard was them deprecating analyzer plugins altogether.

https://github.com/dart-lang/sdk/issues/53402#issuecomment-2487049237

**Comment by jonaslundquist** [comment]
Is there a way to make custom_lint not look at the sub workspaces, only the root package?

**Comment by Rexios80** [comment]
I have precisely zero trust in a Google team to release a feature in a reasonable amount of time. @rrousselGit if I make a PR to add workspace support will you work with me on it? My intuition is this shouldn't be hard to implement.

**Comment by rrousselGit** [comment]
Sure! If you want to rake care of it, that'd be lovely.

I just don't want to spend time on it when I could instead spend time on Freezed/Riverpod.
But that's definitely valuable in the short term

**Comment by Rexios80** [comment]
@rrousselGit The fix was indeed trivial and all tests pass

**Comment by Rexios80** [comment]
Confirmed that `custom_lint: 0.7.5` works in my projects that are set up with Dart workspaces. Thanks for the help with this!

**Comment by rrousselGit** [comment]
I did nothing. _You_ did it :)

**Comment by jogboms** [comment]
How does this work now @Rexios80 ? I couldn't get it to work however I tried. Do you now have a single analysis options file in the root? do you have custom_lint as a dev_dependency for all the packages? what is your current setup that works?

**Comment by Rexios80** [comment]
@jogboms I didn't change anything except updating custom_lint in each project

**Comment by Rexios80** [comment]
@jogboms Did you end up figuring this out? Are you sure you have the latest version of custom_lint?

**Comment by jogboms** [comment]
I still couldn't get it to work and I am on the latest version as well. I have a single analysis options in the root of the project and `pkg:custom_lint` for each package yet it doesn't find any issues while running the command or via the IDE.

**Comment by Rexios80** [comment]
I just tried migrating to a single `analysis_options.yaml` in the workspace root and added the `custom_lint` dependency to the workspace pubspec and it seems to be working. Not sure why it wouldn't be working for you. Maybe try `flutter clean && flutter pub get`?

**Comment by jogboms** [comment]
I’ll give it another go then. Thanks for the insights.

**Comment by Rexios80** [comment]
@jogboms There is definitely an issue with custom_lint in workspaces still

Lints are shown in the editor, but `dart run custom_lint` indeed finds no issues

**Comment by Rexios80** [comment]
@jogboms This should be fixed by https://github.com/invertase/dart_custom_lint/pull/336
